packages/loader/container-loader/src/test/connectionManager.spec.ts

```diff
@@ -192,6 +192,37 @@ describe("connectionManager", () => {
 		);
 	});
 
+	it("reconnectOnError - nack retryAfter", async () => {
+		const connectionManager = createConnectionManager();
+		connectionManager.connect({ text: "test:reconnectOnError" });
+		let connection = await waitForConnection();
+
+		const nack: Partial<INack> = {
+			content: {
+				code: 429,
+				type: NackErrorType.ThrottlingError,
+				message: "throttled",
+				retryAfter: 0.5, // 500 ms
+			},
+		};
+		connection.emitNack("docId", [nack]);
+
+		assert(!closed, "Don't expect closeHandler to be called with retryable Nack");
+		assert(connection.disposed, "Expect connection to be disconnected");
+		assert.strictEqual(disconnectCount, 1, "Expect 1 disconnect from emitting a Nack");
+
+		// Async test we aren't connected within 300 ms
+		let checkedTimeout = false;
+		setTimeout(() => {
```

Comment:
Late to the party, but for a unit test we can usually use Sinon clocks to simulate the passage of time without the test having to actually wait half a second, which is nice since that can add up quickly in CI that is run all the time for the team.